articles/quickstart/webapp/rails/01-login.md

```diff
@@ -9,169 +9,124 @@ topics:
-:::
+### Add an Auth0 controller
+Create an Auth0 controller file in the root of your project's dierctory `./config/controllers/auth0_controller.rb` for handling of the authentication callback.  Inside of the callback method assign the hash of user information returned `request.env['omniauth.auth']` to the active session.
 
 ```ruby
-# app/controllers/auth0_controller.rb
-
+# ./app/controllers/auth0_controller.rb
 class Auth0Controller < ApplicationController
   def callback
-    # This stores all the user information that came from Auth0
-    # and the IdP
+    # This stores all the user information that came from Auth0 and the IdP
     session[:userinfo] = request.env['omniauth.auth']
 
     # Redirect to the URL you want after successful auth
     redirect_to '/dashboard'
   end
 
   def failure
-    # show a failure page or redirect to an error page
+    # Handles failed authentication -- Show a failure page (you can also handle with a redirect)
     @error_msg = request.params['message']
   end
 end
 ```
 
-Replace the generated routes with the following:
+Add the following routes to your `./config/routes.rb` file:
 
 ```ruby
-# config/routes.rb
-
+# ./config/routes.rb
 Rails.application.routes.draw do
-  get 'auth/auth0/callback' => 'auth0#callback'
-  get 'auth/failure' => 'auth0#failure'
+  get '/auth/auth0/callback' => 'auth0#callback'
+  get '/auth/failure' => 'auth0#failure'
 end
 ```
 
-## Trigger Authentication
-
-We need a way for users to trigger authentication. Add a link to `/auth/auth0` anywhere in an existing template or use the steps below to generate a homepage in a new app.
+## Login
+A user can now log into your application by visiting the `/auth/auth0` endpoint.
 
 ::: warning
-To prevent forged authentication requests, make sure that you add a link with a method of `:post` (as described below using the `link_to` function in Rails) or create a form with a CSRF token included.
+To prevent forged authentication requests, use the link_to or button_to helper methods using the `:post` method
 :::
 
-Run the following command to generate the homepage controller and views:
-
-```bash
-rails generate controller home show --skip-assets
+```ruby
+  # Place the login button anywhere
+  ${"<%= button_to 'Login', 'auth/auth0', method: :post %>"}
 ```
 
-Add the following to the generated `show.html.erb` file:
-
-```
-<!-- app/views/home/show.html.erb -->
+### Redirect User to Login
+In order for us to redirect the user to login without any user interaction, we will need to create a non-interactive redirect page.  The redirect page will [prevent forged authentication requests](https://github.com/cookpad/omniauth-rails_csrf_protection) and handle the `:POST` request for us automatically using javascript.
```

Comment:
What's missing from here for me is the reasoning behind _why_ a user might want to do this. We could include a sentence, something like "As an example, you might want to do this when..." just to give an example scenario when this might be useful.

Right now without it, it's a little abstract.

Reply:
Also in general it's a good approach, but the only annoyance I found was that if a user decides to back out of the login page back to their app, they'll hit the redirect page which then automatically forwards them to the login page again. They have to do that thing where they click back twice to get to the right page.

Reply:
@stevehobbsdev if they were to back out of the login page to a protected route, wouldnt they be redirected back to login traditionally?

Reply:
I think we will move this out of the quickstart and into an example